### ISSUE TITLE
Crash on `foldingRange`

### ISSUE THREAD

**Issue by SuperAuguste** [zigtools/zls#957]
[Fuzzing group for this issue ↗](https://fuzzing.zigtools.org/group/872ffdb19c09206f23ddaa492172d2e0) (872ffdb19c09206f23ddaa492172d2e0)

The following pieces of information may be outdated; please go to the fuzzing webpage to obtain newer samples if that is the case.

## Generic error output

```log
panic: start index 370 is larger than end index 354
/opt/hostedtoolcache/zig/zig-linux-x86_64-0.11.0-dev.1507+6f13a725a/x64/lib/std/zig/Ast.zig:1232:31: 0x54922c in tokensOnSameLine (zls)
    const source = tree.source[token_starts[token1]..token_starts[token2]];
                              ^
/home/runner/work/zls/zls/zls/src/folding_range.zig:32:42: 0x4dba2f in add (zls)
        if (builder.tree.tokensOnSameLine(start, end)) return;
                                         ^
/home/runner/work/zls/zls/zls/src/folding_range.zig:184:32: 0x458133 in generateFoldingRanges (zls)
                try builder.add(null, list_start_tok, list_end_tok, .exclusive, .exclusive);
                               ^
/home/runner/work/zls/zls/zls/src/Server.zig:2586:82: 0x3ddc91 in foldingRangeHandler (zls)
    return try folding_range.generateFoldingRanges(allocator, handle.tree, server.offset_encoding);
                                                                                 ^
/home/runner/work/zls/zls/zls/src/Server.zig:2920:37: 0x3961b8 in processMessage (zls)
            const response = handler(server, params) catch |err| {
                                    ^
/home/runner/work/zls/zls/zls/src/Server.zig:2798:26: 0x342633 in processJsonRpc (zls)
    server.processMessage(message) catch |err| switch (message) {
                         ^
/home/runner/work/zls/zls/zls/src/main.zig:89:30: 0x3415c9 in loop (zls)
        server.processJsonRpc(&amp;arena, json_message);
                             ^
/home/runner/work/zls/zls/zls/src/main.zig:402:13: 0x3431cb in main (zls)
    try loop(&amp;server, record_file, replay_file);
            ^
/opt/hostedtoolcache/zig/zig-linux-x86_64-0.11.0-dev.1507+6f13a725a/x64/lib/std/start.zig:616:37: 0x32bdb9 in posixCallMainAndExit (zls)
            const result = root.main() catch |err| {
                                    ^
/opt/hostedtoolcache/zig/zig-linux-x86_64-0.11.0-dev.1507+6f13a725a/x64/lib/std/start.zig:376:5: 0x32b821 in _start (zls)
    @call(.never_inline, posixCallMainAndExit, .{});
    ^

```

## Latest (at time of filling) entry

|  | Zig | ZLS
| --- | --- | --- |
| Versions | 0.11.0-dev.1507+6f13a725a | 0.11.0-dev.153+1b3c3de |
| Files | [Error ↗](https://fuzzing.zigtools.org/entry/1675311686101/stderr.log) | [Input ↗](https://fuzzing.zigtools.org/entry/1675311686101/stdin.log) |
|  | [Output ↗](https://fuzzing.zigtools.org/entry/1675311686101/stdout.log) | [Zig Repro ↗](https://fuzzing.zigtools.org/entry/1675311686101/principal.zig) |

**Comment by SuperAuguste** [comment]
Reproduced locally, pick whichever principal file you'd like :)

**Comment by SuperAuguste** [comment]
Fixed by #958.
